lib/glimmer/tk/toplevel_proxy.rb

```diff
@@ -32,19 +32,43 @@ class ToplevelProxy < WidgetProxy
       DEFAULT_HEIGHT = 0
       
       attr_reader :tk
-      attr_accessor :escapable
+      attr_accessor :escapable, :modal, :centered
       alias escapable? escapable
-      
+      alias modal? modal
+      alias centered? centered
+
       def post_add_content
-        center_within_screen unless @x || @y
+        center_within_screen if centered?
+
         if escapable?
            on('KeyPress') do |event|
-            if event.keysym == 'Escape'
-              grab_release
+            if event.state == 0 && event.keysym == 'Escape'
+              tk.grab_release
               destroy
             end
           end
         end
+
+        if is_a?(Glimmer::Tk::ToplevelProxy) && modal?
+          center_within_root unless centered?
+          root_parent_proxy.withdraw
+          tk.grab_set
+          on('WM_DELETE_WINDOW') do
+            tk.grab_release
+            root_parent_proxy.deiconify
+            destroy
+            true
+          end
+          on('destroy') do
+            tk.grab_release
+            root_parent_proxy.deiconify
+            true
+          end
+        end
+
+        if is_a?(Glimmer::Tk::ToplevelProxy) && @tk.iconphoto.nil? && root_parent_proxy.iconphoto
```

Comment:
I am not sure why you are checking if `is_a?(Glimmer::Tk::ToplevelProxy)`. The code is inside an instance method of `Glimmer::Tk::ToplevelProxy`, so `self` will always have `is_a?(Glimmer::Tk::ToplevelProxy)` as `true`.

Reply:
That's not the way inheritance/polymorphism works though. `RootProxy` extends `ToplevelProxy`, so both `ToplevelProxy` and `RootProxy` will return `true` if you check `is_a?(Glimmer::Tk::ToplevelProxy)` (unless you check if the object.class == `Glimmer::Tk::ToplevelProxy`, which is usually not a good practice except in very rare situations). Only a superclass of `ToplevelProxy` would return `false` for that check, but it wouldn't have that code in the first place since it is the superclass, so then checking that condition would be unnecessary for a superclass (not your case though since you mentioned the subclass `RootProxy`). Maybe, you wanted to check `is_a?(Glimmer::Tk::RootProxy)` as that would return `true` only in `RootProxy` and `false` in a standard `ToplevelProxy`. But, usually checking for a subclass in a superclass is a code smell, so there are better ways of going about it, like by following the [Template Method Design Pattern](https://en.wikipedia.org/wiki/Template_method_pattern) and deferring some of the logic to a method that has an empty implementation in `ToplevelProxy` and a filled in implementation in `RootProxy`.

Here is a GIRB (Glimmer IRB) example demonstrating my point above:
<img width="495" alt="Screenshot 2023-05-13 at 11 04 41 AM" src="https://github.com/AndyObtiva/glimmer-dsl-tk/assets/23052/8c8b27aa-ccd4-4de5-8883-f36a82d1b45d">


In any case, I will accept this code for now, and will update it myself in the future.

Thank you again for your contributions.